### ISSUE TITLE
Pro cursors still failing with grouped relation

### ISSUE THREAD

**Issue by defrex** [rmosolgo/graphql-ruby#576]
Hi Robert.

After upgrading to `1.2.1` an enabling stable cursors, I'm still getting test failures for grouped queries. Below is an example query and the traceback it generates.

```SQL
SELECT `conversations`.* FROM `conversations`  
WHERE (conversations.user_id = 85 OR conversations.recipient_id = 85) 
  AND (conversations.site_id = 2) 
GROUP BY conversations.user_id, conversations.recipient_id
```

```
NoMethodError:
  undefined method `name' for nil:NilClass
# /usr/local/bundle/gems/graphql-pro-1.2.1/lib/graphql/pro/relation_connection/order.rb:57:in `normalize'
# /usr/local/bundle/gems/graphql-pro-1.2.1/lib/graphql/pro/relation_connection.rb:10:in `new'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/relay/connection_resolve.rb:26:in `build_connection'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/relay/connection_resolve.rb:18:in `call'
# /usr/local/bundle/gems/graphql-pro-1.2.1/lib/graphql/pro/access/instrumentation/authorized_resolve.rb:21:in `call'
# /usr/local/bundle/gems/graphql-pro-1.2.1/lib/graphql/pro/monitoring/eager_load_relation.rb:13:in `call'
# /usr/local/bundle/gems/graphql-pro-1.2.1/lib/graphql/pro/monitoring/new_relic_platform.rb:16:in `block in call'
# /usr/local/bundle/gems/newrelic_rpm-3.14.0.305/lib/new_relic/agent/method_tracer_helpers.rb:82:in `trace_execution_scoped'
# /usr/local/bundle/gems/graphql-pro-1.2.1/lib/graphql/pro/monitoring/new_relic_platform.rb:15:in `call'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/field.rb:196:in `resolve'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/define/instance_definable.rb:179:in `block (3 levels) in ensure_defined'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/execution/execute.rb:205:in `call'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/schema/middleware_chain.rb:45:in `invoke_core'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/schema/middleware_chain.rb:38:in `invoke'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/execution/execute.rb:61:in `resolve_field'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/execution/execute.rb:31:in `block in resolve_selection'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/internal_representation/selection.rb:41:in `block in each_selection'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/internal_representation/selection.rb:38:in `each'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/internal_representation/selection.rb:38:in `each_selection'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/execution/execute.rb:29:in `resolve_selection'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/execution/execute.rb:170:in `resolve_value'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/execution/execute.rb:105:in `continue_resolve_field'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/execution/execute.rb:76:in `resolve_field'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/execution/execute.rb:31:in `block in resolve_selection'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/internal_representation/selection.rb:41:in `block in each_selection'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/internal_representation/selection.rb:38:in `each'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/internal_representation/selection.rb:38:in `each_selection'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/execution/execute.rb:29:in `resolve_selection'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/execution/execute.rb:9:in `execute'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/query/executor.rb:37:in `execute'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/query/executor.rb:18:in `result'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/query.rb:115:in `result'
# /usr/local/bundle/gems/graphql-1.4.4/lib/graphql/schema.rb:184:in `execute'
# ./spec/support/graph.rb:13:in `graph_query'
# ./spec/graph/graph_types/conversation_spec.rb:289:in `block (3 levels) in <top (required)>'
```

**Comment by rmosolgo** [comment]
Hmm, there are two issues: 

- The connection is assuming that there's at least one order clause, but there isn't an order clause, so :boom: 
- For cursors to be stable, they _must_ be sorted by a unique value. For non-grouped relations, we "automatically" add an order-by-primary-key. For grouped relations, we can't guess a unique column, so one must be provided by the application. This is gently mentioned [in the docs](http://rmosolgo.github.io/graphql-ruby/pro/cursors#grouped-relations), but obviously it's not doing its job! 

So, how about this approach: If a grouped relation doesn't have _any_ orderings, raise a helpful error so that application developers can add a unique order clause. (If it _does_ have orderings, continue to assume that it's unique.)

Does that seem like a good fit?

**Comment by defrex** [comment]
That seems perfect. I don't mind adding an order, I just didn't realize that was the problem.

**Comment by defrex** [comment]
I can confirm that order solves the issue.

**Comment by rmosolgo** [comment]
Sweet, released in 1.2.2! #577
